plugin/discovery/get.go

```diff
@@ -93,6 +104,18 @@ func (i *ProviderInstaller) Get(provider string, req Constraints) (PluginMeta, e
 	// take the first matching plugin we find
 	for _, v := range versions {
 		url := providerURL(provider, v.String())
+
+		if !i.SkipVerify {
+			sha256, err := getProviderChecksum(provider, v.String())
+			if err != nil {
+				return PluginMeta{}, nil
+			}
+
+			if sha256 != "" {
+				url = url + "?checksum=sha256:" + sha256
```

Comment:
Is this extra argument an S3 feature? Or is this just a no-op extra argument?

(Either way, I think it could do with a comment explaining why we're doing this.)